add support for codex animal, weapon, reliquary

ShigemoriHakura opened the pull request:
## Description

add support for codex animal, weapon, reliquary
fix the error when running in idea
fix the translation error in givechar

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Issues fixed by this PR

<!--- Put the links of issues that may be fixed by this PR here (if any). -->
## Type of changes

<!--- Put an `x` in all the boxes that apply your changes. -->

- [x] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

ShigemoriHakura committed:
add support for codex animal, weapon, reliquary

add support for codex
fix the error when running in idea
fix the translation error in givechar
QueryCodexMonsterBeKilledNumRsp still has some problem

arub3do commented:
I guess these `CmdId` is based on V2.2.0?
```
Proto.QueryCodexMonsterBeKilledNumReq,4201,
Proto.QueryCodexMonsterBeKilledNumRsp,4210,
```

Maybe V2.6 should be: 4207/4208

ShigemoriHakura commented:
> I guess these `CmdId` is based on V2.2.0?
> 
> ```
> Proto.QueryCodexMonsterBeKilledNumReq,4201,
> Proto.QueryCodexMonsterBeKilledNumRsp,4210,
> ```
> 
> Maybe V2.6 should be: 4207/4208

The problem is that only the first animal in QueryCodexMonsterBeKilledNumReq's codexIdList can show the correct beKilledNum. I think it's not due to the cmdid mismatch.

ShigemoriHakura committed:
Fix PacketQueryCodexMonsterBeKilledNumRsp

ShigemoriHakura committed:
Merge remote-tracking branch 'upstream/development' into development_CodexWeapon

Melledy commented on src/main/java/emu/grasscutter/utils/Utils.java:
```diff
@@ -200,8 +201,10 @@ public static void startupCheck() {
 		// Make sure the data folder is populated, if there are any missing files copy them from resources
-		DataLoader.CheckAllFiles();
-
+		if(Grasscutter.class.getResource("").getProtocol() == "jar"){
```
Could you explain this change?

ShigemoriHakura replied:
DataLoader.CheckAllFiles() only works when running in jar (grasscutter-xxx-dev.jar),

Otherwise it will throw out an error:
java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "filenames" is null
	at emu.grasscutter.data.DataLoader.CheckAllFiles(DataLoader.java:58)
	at emu.grasscutter.utils.Utils.startupCheck(Utils.java:208)
	at emu.grasscutter.Grasscutter.<clinit>(Grasscutter.java:77)

Can be reproduced by using .\gradlew.bat run

HotaruYS replied:
I think this needs more attention and be fixed properly in separate PR. Your fix will just ensure it won't crash but it won't copy necessary files. Also the log message is pointless.

ShigemoriHakura replied:
I know, but I think those who uses ide to run grasscutter know how to copy the data from resources to /data

TalexDreamSoul commented:
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

ShigemoriHakura committed:
Refactor codex files and add support for achievements

Added an event system to check achievement triggers
Refactor codex files to match codestyle
Fix proto mismatch (TakeAchievementRewardReq, TakeAchievementRewardRsp)

ShigemoriHakura commented:
Refactoring
